src/Stack/PackageDump.hs

```diff
@@ -64,7 +64,7 @@ ghcPkgCmdArgs pkgexe@(GhcPkgExe pkgPath) cmd mpkgDbs sink = do
     case reverse mpkgDbs of
         (pkgDb:_) -> createDatabase pkgexe pkgDb -- TODO maybe use some retry logic instead?
         _ -> return ()
-    sinkProcessStdout (toFilePath pkgPath) args sink'
+    snd <$> sinkProcessStderrStdout (toFilePath pkgPath) args CL.sinkNull sink'
```

Comment:
Just to be clear, this changes the behavior of calling the process. Previously, `stderr` from the process would be inherited and therefore error output would show up in the console. Now it will be swallowed. Not terrible, but it would be ideal to avoid that if possible.

Reply:
The existing comment at `sinkProcessStdout` warns:
~~~
-- | Consume the stdout of a process feeding strict 'ByteString's to a consumer.
-- If the process fails, spits out stdout and stderr as error log
-- level. Should not be used for long-running processes or ones with
-- lots of output; for that use 'sinkProcessStderrStdout'.
~~~
For some reason, the use of `sinkProcessStdout` on macOS had begun to generate `startProcess: posix_spawnp: illegal operation (Inappropriate ioctl for device)` errors. Substituting `sinkProcessStderrStdout` caused them to disappear.

Reply:
I have filed a bug report here: https://github.com/haskell/process/issues/251

@mpilgrem I have been getting various version of the errors, ranging from:

```
startProcess: posix_spawnp: invalid argument (Bad file descriptor)
startProcess: posix_spawnp: does not exist (No such file or directory)
```

I haven't seen the `ioctl` error message that you are seeing, but it is a concern.